diff -up umbrello-18.11.90/umbrello/main.cpp.omv~ umbrello-18.11.90/umbrello/main.cpp
--- umbrello-18.11.90/umbrello/main.cpp.omv~	2018-12-06 13:15:56.750875370 +0100
+++ umbrello-18.11.90/umbrello/main.cpp	2018-12-06 13:17:00.635058580 +0100
@@ -19,9 +19,7 @@
 
 // kde includes
 #if QT_VERSION > 0x050000
-namespace dummy {
 #include <kaboutdata.h>
-}
 #include <k4aboutdata.h>
 #define KAboutData K4AboutData
 #include <KCrash>
