diff -up umbrello-20.03.90/CMakeLists.txt.1~ umbrello-20.03.90/CMakeLists.txt
--- umbrello-20.03.90/CMakeLists.txt.1~	2020-04-05 18:00:13.328560984 +0200
+++ umbrello-20.03.90/CMakeLists.txt	2020-04-05 18:01:50.468859584 +0200
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 2.8.12 FA
 
 project(umbrello)
 
+set(CMAKE_CXX_STANDARD 17)
+
 if(POLICY CMP0028)
     cmake_policy(SET CMP0028 OLD)
 endif()
diff -up umbrello-20.03.90/unittests/CMakeLists.txt.1~ umbrello-20.03.90/unittests/CMakeLists.txt
--- umbrello-20.03.90/unittests/CMakeLists.txt.1~	2020-03-28 23:41:36.000000000 +0100
+++ umbrello-20.03.90/unittests/CMakeLists.txt	2020-04-05 18:00:13.328560984 +0200
@@ -181,7 +181,7 @@ if(LLVM_FOUND AND (Clang_FOUND OR CLANG_
     # Find the libraries that correspond to the LLVM components
     # that we wish to use
     if(LLVM_PACKAGE_VERSION VERSION_GREATER_EQUAL "7.0.0")
-        set(llvm_libs LLVM)
+        set(llvm_libs LLVMSupport LLVMCore LLVMIRReader LLVMAnalysis)
     else()
         llvm_map_components_to_libnames(llvm_libs support core irreader analysis)
     endif()
diff -up umbrello-20.03.90/unittests/testllvmparser.cpp.1~ umbrello-20.03.90/unittests/testllvmparser.cpp
--- umbrello-20.03.90/unittests/testllvmparser.cpp.1~	2020-03-28 23:41:36.000000000 +0100
+++ umbrello-20.03.90/unittests/testllvmparser.cpp	2020-04-05 18:00:13.328560984 +0200
@@ -103,6 +103,6 @@ int main(int argc, char **argv)
             return 1;
         defaultSource = source.readAll();
     }
-    return clang::tooling::runToolOnCode(new FindNamedClassAction, defaultSource.data()) ? 0 : 2;
+    return clang::tooling::runToolOnCode(std::make_unique<FindNamedClassAction>(), defaultSource.data()) ? 0 : 2;
 }
 
